main.rb

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env ruby
+require 'base64'
+require 'fileutils'
+require 'time'
+require 'open3'
+require 'colored'
+require 'English'
+require 'shellwords'
+
+def abort_with_message(msg)
+  abort "@@[error] #{msg}"
+end
+
+def env_has_key(key)
+  value = ENV[key]
+  value.nil? || value.strip.empty? ? abort_with_message("Missing #{key}") : value
+end
+
+def get_env_variable(key)
+  value = ENV[key]
+  value && !value.strip.empty? ? value : nil
+end
+
+def param_checker(*params)
+  params.all? { |p| !p.to_s.strip.empty? }
+end
+
+def validate_input_script_folder(input_path, script_file)
+  file_path = File.join(input_path, script_file)
+  abort_with_message("Script file or repository directory not found at AC_SCRIPT_REPO_DIR: #{file_path}") unless File.exist?(file_path)
+end
+
+def run_command(cmd)
+  masked = cmd.gsub(/(Authorization:\s*\w+\s+)\S+/,'\1********')
+  puts "@@[command] #{masked}"
+  stdout, stderr, status = Open3.capture3(cmd)
+  puts stdout unless stdout.empty?
+  unless status.success?
+    raise "Command failed (#{status.exitstatus}):\n#{stderr}"
+  end
+  true
+end
+def get_path_clone_repo(clone_url, extra_header = nil)
+  root = "Cloned_Script_#{Time.now.strftime('%Y%m%d_%H%M%S')}"
+  FileUtils.mkdir_p(root)
+  repo = File.basename(clone_url, '.git')
+  path = File.join(root, repo)
+  FileUtils.cd(root) do
+    cmd = extra_header ? %Q[git -c "#{extra_header}" clone #{clone_url}] : "git clone #{clone_url}"
+    abort("Error: Cloning failed") unless run_command(cmd)
+  end
+  path
+end
+
+def prepare_args(raw_params)
+  parts = Shellwords.split(raw_params.to_s.gsub(',', ' '))
+  parts.empty? ? '' : " #{parts.join(' ')}"
+end
+
+def execute_script_in(folder, script_file, branch, script_args)
+  FileUtils.cd(folder) do
+    if branch.nil? || branch.strip.empty?
+      branch = "main"
+    end
+    abort_with_message("Git checkout failed: #{branch}") unless run_command("git checkout #{branch}")
+    abort_with_message("Script file not found: #{script_file}") unless File.exist?(script_file)
+    extname = File.extname(script_file).downcase
+    case extname
+    when '.sh'
+      FileUtils.chmod('+x', script_file)
+      run_command("./#{script_file}#{script_args}")
+    when '.py'
+      run_command("python #{script_file}#{script_args}")
+    when '.rb'
+      run_command("ruby #{script_file}#{script_args}")
+    when '.pl'
+      run_command("perl #{script_file}#{script_args}")
+    when '.js'
+      run_command("node #{script_file}#{script_args}")
+    when '.java'
+      run_command("javac #{script_file}")
+      class_name = File.basename(script_file, '.java')
+      run_command("java #{class_name}#{script_args}")
+    else
+      abort_with_message("Unsupported script type: #{script_file}")
+    end
+  end
+end
+
+def write_env_file(root_folder)
+  File.open(ENV['AC_ENV_FILE_PATH'], 'a') do |f|
+    f.puts "AC_SCRIPT_REPO_OUTPUT_DIR=#{root_folder}"
+  end
+end
```

Comment:
1. If this logic is being placed into a function, it would be better to pass both the variable name (`AC_SCRIPT_REPO_OUTPUT_DIR`) and its value (`root_folder`) as parameters. This makes the function more reusable and decouples it from hardcoded values.
2. Get `AC_ENV_FILE_PATH` with `env_has_key` func
```suggestion
def write_env_file(env_var_name, value)
  File.open(env_has_key("AC_ENV_FILE_PATH"), 'a') do |f|
    f.puts "#{env_var_name}=#{value}"
  end
end
```